src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -43,6 +43,49 @@
 #endif
 
 
+///////////////////////////////////////////////////////////////////////
+/* SELFMADE -  Implementation of itoa() */
+
+char* myitoa(int num, char* str, int base) {
+  int i = 0;
+  bool isNegative = false;
+  /* Handle 0 explicitely, otherwise empty string is printed for 0 */
+  if (num == 0) {
+    str[i++] = '0';
+    str[i] = '\0';
+    return str;
+  }
+  // In standard itoa(), negative numbers are handled only with
+  // base 10. Otherwise numbers are considered unsigned.
+  if (num < 0 && base == 10) {
+    isNegative = true;
+    num = -num;
+  }
+  // Process individual digits
+  while (num != 0) {
+    int rem = num % base;
+    str[i++] = (rem > 9) ? (rem - 10) + 'a' : rem + '0';
+    num = num / base;
+  }
+
+  if (isNegative) str[i++] = '-';  // If number is negative, append '-'
+  str[i] = '\0';                   // Append string terminator
+  myreverse(str, i);               // Reverse the string
```

Comment:
Ich bin jetzt schon eine weile am überlegen, ob man die Reverse Sache nicht auflösen könnte.
Die Anzahl an Zeichen ist ja durch num vorgegeben. bei int wären das maximal 6 Zeichen mit dem null term wären wir bei 7 Byte Speicherbedarf.

Würde man Zeile 67 ändern in
`str[i--] = (rem > 9) ? (rem - 10) + 'a' : rem + '0'; `würde sich der char array von hinten aufgebaut werden.
Alles in allem hätten wir dann leerzeichen am Anfang, die wir natürlich nicht wollen.
Dazu bräuchte es am Ende in memmove um die chars an den Anfang des char arrays zu schieben.

Vielleicht hast Du darüber auch bereits nachgedacht oder es probiert und verworfen?

Reply:
Wenn wir vorher wüssten, wieviel Stellen die Zahl in "int num" hat, könnte man auch gleich von links aufdröseln. Hast du da eine Idee, wie man das kurz und knapp lösen kann?

Reply:
Kurz und Knapp vielleicht nicht aber so
```
int numDigits(int x)  
{  
    x = abs(x);  // Effizentere Variante hier einsetzen, nur zu Demo Zwecken abs verwendet
    return (x < 10 ? 1 :   
        (x < 100 ? 2 :   
        (x < 1000 ? 3 :   
        (x < 10000 ? 4 :   
        5))));  
}  
```
alternativ ginge auch eine Schleife, aber das dürfte Aufwändiger in der Berechnung sein
```
  uint8_t digits = 0; do { x/= 10; digits++; } while (x != 0);
```

Reply:
Ich habe da eine interessante Diskussion gefunden: https://forum.arduino.cc/index.php?topic=634400.0